datafusion/functions-array/src/string.rs

```diff
@@ -281,6 +283,49 @@ pub(super) fn array_to_string_inner(args: &[ArrayRef]) -> Result<ArrayRef> {
 
                 Ok(arg)
             }
+            Dictionary(_, _) => {
+                let any_dict_array = arr.as_any_dictionary_opt().ok_or_else(|| {
```

Comment:
Thank you @EduardoVega  -- it is very impressive you got this to work

I am somewhat concerned that this is quite a bit of code (it actually handles dictionaries of any type, not just string dictionaries, and that it also supports arbitrary key types -- not just the integer types specified)

I am going to try and simplify this a bit

Reply:
I pushed 5a5397bef that I think simplifies the implementation a bit -- let me know what you think